pkg/controller/nodemaintenance/nodemaintenance_controller.go

```
@@ -55,268 +131,477 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
+		return reconcile.Result{Requeue: true, RequeueAfter: RequeuDrainingWaitTime * time.Second}, nil
+	}
+
+	statusAnnoData.Status = NodeStateMaintenanceMode
+	r.setStatusAnnotation(node, statusAnnoData)
+
+	return reconcile.Result{}, nil
 }
 
-var Handler ReconcileHandler
+func (r *ReconcileNodeMaintenance) processMaintModeOff(statusAnnoData *ReconcileNodeMaintenanceStatusInfo, node *corev1.Node) (reconcile.Result, error) {
 
-// Reconcile reads that state of the cluster for a NodeMaintenance object and makes changes based on the state read
-// and what is in the NodeMaintenance.Spec
-// Note:
-// The Controller will requeue the Request to be processed again if the returned error is non-nil or
-// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	reqLogger := log.WithFields(log.Fields{"Request.Namespace": request.Namespace, "Request.Name": request.Name})
-	reqLogger.Info("Reconciling NodeMaintenance")
+	nodeName := node.ObjectMeta.Name
 
-	// Fetch the NodeMaintenance instance
-	instance := &kubevirtv1alpha1.NodeMaintenance{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	drainer, _, err := r.initDrainer()
 	if err != nil {
-		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			reqLogger.Infof("NodeMaintenance Object: %s Deleted ", request.NamespacedName)
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		reqLogger.Info("Error reading the request object, requeuing.")
-		return reconcile.Result{}, err
+		return reconcile.Result{Requeue: true, RequeueAfter: RequeuDrainingWaitTime * time.Second}, nil
```

Don't we want to return the error too?